addons/mrp/tests/test_bom.py

```diff
@@ -31,6 +31,37 @@ def test_01_explode(self):
             set([line[0].id for line in lines]),
             set((self.bom_2 | self.bom_3).mapped('bom_line_ids').filtered(lambda line: not line.child_bom_id or line.child_bom_id.type != 'phantom').ids))
 
+    def test_02_explode_rounding(self):
+        fns = self.env['product.product'].create({'name': 'FNS', 'type': 'product'})
+        cmp1 = self.env['product.product'].create({'name': 'CMP1', 'type': 'product'})
+        cmp2 = self.env['product.product'].create({'name': 'CMP2', 'type': 'product'})
+        self.uom_unit.rounding = 0.01
+
+        fns_bom = self.env['mrp.bom'].create({
+            'product_id': fns.id,
+            'product_tmpl_id': fns.product_tmpl_id.id,
+            'product_uom_id': self.uom_unit.id,
```

Comment:
```suggestion
            'product_tmpl_id': fns.product_tmpl_id.id,
```
- Since there are no variant, the template is enough.
- If the uom is the default one, no need to set it.